src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -979,23 +979,18 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
       specMember
     }
 
-    if (sym.isMethod) {
-      if (hasUnspecializableAnnotation(sym)) {
-        List()
-      } else {
-        val stvars = specializedTypeVars(sym)
-        if (stvars.nonEmpty)
-          debuglog("specialized %s on %s".format(sym.fullLocationString, stvars.map(_.name).mkString(", ")))
+    if (!sym.isMethod || sym.isConstructor || hasUnspecializableAnnotation(sym)) {
+      Nil
+    } else {
+      val stvars = specializedTypeVars(sym)
+      if (stvars.nonEmpty)
+        debuglog("specialized %s on %s".format(sym.fullLocationString, stvars.map(_.name).mkString(", ")))
 
-        val tps1 = if (sym.isConstructor) tps filter (sym.info.paramTypes contains _) else tps
```

Comment:
Looks like constructors are handled elsewhere

Reply:
I tried to retrace the code/data flow that establishes that constructors are not seen here, but couldn't find the spot where they are excluded (transformInfo -> specializeClass -> normalizeMember -> specializeMember). Could you elaborate?

Reply:
I'm not sure either. But `sym.info.paramTypes contains _` would always return `false` because it checks if a list of types contains a symbol. Trying to fix it by using `sym.info.paramTypes.map(_.typeSymbol)` instead leads to the following:
```scala
scala> case class Foo[@specialized(Int) T](bar: T)
                                          ^
       error: constructor Foo$mcI$sp is defined twice;
         the conflicting constructor Foo$mcI$sp was defined at line 1:36
```

Reply:
Aha! I missed the `contains` gotcha even though that was the theme of the PR 😅